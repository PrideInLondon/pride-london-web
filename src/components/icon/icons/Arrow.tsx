import React from 'react'
import { InternalIconProps } from '../Icon.types'

const Arrow = ({ colors: { primary }, ...props }: InternalIconProps) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    viewBox="0 0 32 32"
    {...props}
  >
    <g filter="url(#prefix__filter0_d)">
      <path
        fill={primary}
        fillRule="evenodd"
        d="M22.586 14.707l-6.293-6.293L17.707 7l8 8 .707.707-.707.707-8 8L16.293 23l6.293-6.293H5v-2h17.586z"
        clipRule="evenodd"
      />
    </g>
    <defs>
      <filter
        id="prefix__filter0_d"
        width="21.414"
        height="19.414"
        x="5"
        y="7"
        colorInterpolationFilters="sRGB"
        filterUnits="userSpaceOnUse"
      >
        <feFlood floodOpacity="0" result="BackgroundImageFix" />
        <feColorMatrix
          in="SourceAlpha"
          values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
        />
        <feOffset dy="2" />
        <feColorMatrix values="0 0 0 0 0.172549 0 0 0 0 0.854902 0 0 0 0 0.615686 0 0 0 1 0" />
        <feBlend in2="BackgroundImageFix" result="effect1_dropShadow" />
        <feBlend in="SourceGraphic" in2="effect1_dropShadow" result="shape" />
      </filter>
    </defs>
  </svg>
)

export default Arrow
