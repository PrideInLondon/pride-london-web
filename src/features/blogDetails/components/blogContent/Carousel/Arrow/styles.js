import styled from 'styled-components'
import theme from '../../../../../../theme/theme'

export const ArrowContainer = styled.button`
  position: absolute;
  z-index: 1;
  top: 50%;
  ${({ left }) => (left ? 'left: 0;' : 'right: 0;')}
  transform: translateY(-50%);
  display: flex;
  justify-content: center;
  align-items: center;
  width: 60px;
  height: 60px;
  border: none;
  outline: none;
  border-radius: 4px;
  background-color: ${theme.colors.white};
  & > svg {
    margin: 0 !important;
    display: block;
    ${({ left }) => left && 'transform: rotate(180deg);'}
    width: 12px;
    height: 19px;
  }
`
