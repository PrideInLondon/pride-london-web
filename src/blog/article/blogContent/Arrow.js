import React from 'react'
import PropTypes from 'prop-types'
import { ArrowContainer } from './Arrow.styles'

const Arrow = ({ left, children, onClick }) => {
  return (
    <ArrowContainer type="button" onClick={onClick} left={left}>
      {children}
    </ArrowContainer>
  )
}

Arrow.propTypes = {
  children: PropTypes.node.isRequired,
  left: PropTypes.bool,
  onClick: PropTypes.func,
}

Arrow.defaultProps = {
  left: false,
  onClick: () => {},
}

export default Arrow
