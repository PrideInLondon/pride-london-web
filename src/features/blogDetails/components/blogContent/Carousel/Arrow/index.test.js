import React from 'react'
import { shallow } from 'enzyme'
import Arrow from '.'

describe(Arrow.name, () => {
  it('renders with default values', () => {
    const wrapper = shallow(<Arrow>></Arrow>)
    expect(wrapper).toMatchSnapshot()
  })
})
