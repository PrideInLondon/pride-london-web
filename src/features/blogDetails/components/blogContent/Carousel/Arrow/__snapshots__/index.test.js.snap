// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow renders with default values 1`] = `
<styles__ArrowContainer
  left={false}
  onClick={[Function]}
  type="button"
>
  &gt;
</styles__ArrowContainer>
`;
